Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -0,0 +1,203 @@
+    // Read list of back-transform diag parameters requested by the user //
+    amrex::Print() << " in read parameters for BTD \n";
+
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE( ( warpx.do_back_transformed_diagnostics==true),
+        "the do_back_transformed_diagnostics flag must be set to true for BTDiagnostics");
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE( warpx.gamma_boost > 1.0_rt,
+        "gamma_boost must be > 1 to use the back-transformed diagnostics");
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE( warpx.boost_direction[2] == 1,
+        "The back transformed diagnostics currently only works if the boost is in the z-direction");
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE( warpx.do_moving_window,
+           "The moving window should be on if using the boosted frame diagnostic.");
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE( warpx.moving_window_dir == 2,
+           "The boosted frame diagnostic currently only works if the moving window is in the z direction.");
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+        m_format == "plotfile" || m_format == "openpmd",
+        "<diag>.format must be plotfile or openpmd for back transformed diagnostics");
+    amrex::ParmParse pp(m_diag_name);
+
+    m_file_prefix = "diags/lab_frame_data/" + m_diag_name;
+    pp.query("file_prefix", m_file_prefix);
+    pp.query("do_back_transformed_fields", m_do_back_transformed_fields);
+    pp.query("do_back_transformed_particles", m_do_back_transformed_particles);
+    AMREX_ALWAYS_ASSERT(m_do_back_transformed_fields or m_do_back_transformed_particles);
+
+    pp.get("num_snapshots_lab", m_num_snapshots_lab);
+    m_num_buffers = m_num_snapshots_lab;
+
+    // Read either dz_snapshots_lab or dt_snapshots_lab
+    bool snapshot_interval_is_specified = 0;
+    amrex::Real m_dz_snapshots_lab = 0.0_rt;
+    snapshot_interval_is_specified = pp.query("dt_snapshots_lab", m_dt_snapshots_lab);
+    if ( pp.query("dz_snapshots_lab", m_dz_snapshots_lab) ) {
+        m_dt_snapshots_lab = m_dz_snapshots_lab/PhysConst::c;
+        snapshot_interval_is_specified = 1;
+    }
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(snapshot_interval_is_specified,
+        "For back-transformed diagnostics, user should specify either dz_snapshots_lab or dt_snapshots_lab");
+
+    m_map_actual_fields_to_dump.resize(m_varnames.size());
+    for (int i=0; i < m_varnames.size(); ++i) {
+        m_map_actual_fields_to_dump[i] = m_possible_fields_to_dump[ m_varnames[i] ];
+    }
+
+}
+
+void
+BTDiagnostics::writeMetaData ()
+{
+    // This function will have the same functionality as writeMetaData in
+    // previously used BackTransformedDiagnostics class to write
+    // back-transformed data in a customized format
+}
+
+bool
+BTDiagnostics::DoDump (int step, bool force_flush)
+{
+    // always set to true for BTDiagnostics since back-transform buffers are potentially
+    // filled every timstep. They will be flushed if m_buffer_counter for
+    // the snapshot is equal to m_buffer_size
+    return true;
```

Comment:
I think the spirit of this function is more something like
```c++
BTDiagnostics::DoDump(buffer)
if (buffer.is_full()) || (force_flush){
    return true;
}
return false;
```

Reply:
done.